### PR TITLE
Add a `repeat` option for extractors to WebsiteAgent

### PR DESCRIPTION
**Pull request by knu**
This is useful such as when you want to include the title of a page in all events created from the page.  In combination with the `template` option (#1743) and the `_loop_` variable (#1768), I guess you could finally achieve crawling of paginated contents properly, by adding the link to the next page to the last event created by a check, incrementing the nest level, feeding it back to the agent via a TriggerAgent with conditions on the presence of the said link and the nest level.

The implementation with regard to Repeater may be too tricky, and the wording of documentation seemed far beyond me, and I'm not 100% certain as to the naming of the option, so any feedback is welcome!

### CODE REVIEW

File: app/models/agents/website_agent.rb [hunk 1]
```diff
@@ -42,7 +44,8 @@ class WebsiteAgent < Agent
           "extract": {
             "url": { "css": "#comic img", "value": "@src" },
             "title": { "css": "#comic img", "value": "@title" },
-            "body_text": { "css": "div.main", "value": "string(.)" }
+            "body_text": { "css": "div.main", "value": "string(.)" },
+            "page_title": { "css": "title", "value": "string(.)" }
```
Comment (cantino) [inline app/models/agents/website_agent.rb]: Should this have repeat?

Comment (knu) [inline app/models/agents/website_agent.rb]: Oh yes, that's what I intended.  I'll fix it.

**Comment by knu** [comment]
This allows user to include a value that only appears once in a resource in all events created from it, for _all_ extraction types, which has been possible only in the html/xml extraction types owing to the power of XPath, but never in regexp/json.

@cantino What do you think?

File: app/models/agents/website_agent.rb [hunk 2]
```diff
@@ -379,21 +382,24 @@ def handle_data(body, url, existing_payload)
...
         end
 
-      if output.each_value.each_cons(2).any? { |m, n| m.size != n.size }
-        raise "Got an uneven number of matches for #{interpolated['name']}: #{interpolated['extract'].inspect}"
-      end
-
-      num_tuples = output.each_value.first.size
+      num_tuples = output.each_value.inject(nil) { |num, value|
+        case size = value.size
+        when Float::INFINITY
```
Comment (cantino) [inline app/models/agents/website_agent.rb]: When does this happen?

Comment (knu) [inline app/models/agents/website_agent.rb]: This is what Repeater#size returns, meaning it is an infinite list.  `num_tuples` is being set to a finite number that is the size of finite lists, so it is counted out here.

**Pull request merged**